lib/file-helper.js

```diff
@@ -170,22 +173,23 @@ module.exports = {
   getComponentsData,
   getDirectories,
   getFiles,
-  getFilesByDirectory,
   getFullPageExamples,
-  getListing
+  getListing,
+  listingToArray
 }
 
 /**
  * Directory listing
  *
- * @typedef {Map<string, { path: string; stats: import('fs').Stats }>} DirectoryListing
+ * @typedef {Map<string, DirectoryListingEntry>} DirectoryListing
  */
 
 /**
  * Directory listing entry
  *
  * @typedef {object} DirectoryListingEntry
  * @property {string} path - Relative path to file or directory
+ * @property {DirectoryListing} [entries] - Child directory listing for entry
```

Comment:
Marked optional `[entries]` as a directory listing includes both files and directories

Only directories (not files) will have the `.entries` property available

```mjs
// Listing for components directory
const listing = await getListing(configPaths.components)

// Accordion directory entry
const accordion = listing.get('accordion')

// Accordion files
const accordionEntries = accordion.entries

// Accordion example files
accordionEntries.get('accordion.mjs')
accordionEntries.get('accordion.yaml')
accordionEntries.get('README.md')
```

Reply:
Swapped with `glob()` as discussed in https://github.com/alphagov/govuk-frontend/pull/2946#discussion_r1007203034